### ISSUE TITLE
TryFrom* methods shouldn't throw ArgumentException

### ISSUE THREAD

**Issue by hkrotsik** [ardalis/SmartEnum#95]
In current implementation `TryFromName` and `TryFromValue` throw an ArgumentException when value is an empty string or null. This behavior makes those methods pretty unusable as you have to write a lot of boilerplate code to handle all possible cases.
Also, this behavior differs from all other [TryParse](https://github.com/dotnet/runtime/blob/master/src/libraries/System.Private.CoreLib/src/System/Double.cs#L324) methods with a similar semantics

**Comment by ardalis** [comment]
I agree that any `TryXXX` method should work without exceptions (and just return false if it fails).
